### PR TITLE
[BEAM-5458] Fix python tests

### PR DESCRIPTION
**Pull request by RobbeSneyders**
Fix Python 3 tests which are currently failing, see #6464 

@pabloem @tvalentyn 

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by tvalentyn** [comment]
LGTM

File: sdks/python/setup.py [hunk 1]
```diff
@@ -121,6 +121,7 @@ def get_version():
     # grpcio 1.8.1 and above requires protobuf 3.5.0.post1.
     'protobuf>=3.5.0.post1,<4',
     'pydot>=1.2.0,<1.3',
+    'pyhamcrest>=1.9,<2.0',
```
[the remark this answers is not in the document]
Comment (RobbeSneyders) [reply sdks/python/setup.py]: This shouldn't be here, removed.

**Comment by aaltay** [comment]
I will merge this after tests pass.

**Comment by lostluck** [comment]
Thanks for being on this! Got to it before I could file a jira.

**Comment by RobbeSneyders** [comment]
Python 3 tests should be fixed, but docs are still failing.

**Comment by tweise** [comment]
The JIRA for this is: https://issues.apache.org/jira/browse/BEAM-5458

**Comment by udim** [comment]
There's a lint error:
```
10:36:26 ************* Module apache_beam.runners.portability.fn_api_runner
10:36:26 C:460, 0: Wrong hanging indentation (remove 8 spaces).
10:36:26                                 impulse_pc.encode('utf-8')),
10:36:26                         |       ^ (bad-continuation)
```

**Comment by RobbeSneyders** [comment]
Pylint is fixed.
I don't see why docs are failing. The Sphinx and pyhamcrest versions being installed are still the same as when the tests were passing, and I don't see any changes on the relevant files.

**Comment by udim** [comment]
I think the issue is that the Hamcrest docs website is an external dependency:
https://github.com/apache/beam/blob/da9432b197207ab39edc6cb720286a81153b6d3b/sdks/python/scripts/generate_pydoc.sh#L116

Compare:
https://pyhamcrest.readthedocs.io/en/latest/core/ (empty)
with:
https://pyhamcrest.readthedocs.io/en/v1.9.0/core/

I suggest changing the above line to:
```
  'hamcrest': ('https://pyhamcrest.readthedocs.io/en/stable/', None),
```

**Comment by tweise** [comment]
Let's merge before it breaks again :)

**Pull request merged**